test/Scanner/MethodScannerTest.php

```diff
@@ -110,4 +111,27 @@ public function testMethodScannerWorksWithSingleAbstractFunction()
 
         self::assertTrue($method->isAbstract());
     }
+
+    /**
+     * @expectedException \Zend\Code\Exception\InvalidArgumentException
+     * @expectedExceptionMessage Invalid visibility argument passed to setVisibility.
+     */
+
+    public function testMethodScannerSetVisibilityThrowsInvalidArgumentException()
+    {
+        $methodScanner = new MethodScanner([]);
+
+        $invalidArgument = 42;
+        self::assertTrue(! in_array($invalidArgument, [T_PUBLIC, T_PROTECTED, T_PRIVATE]));
```

Comment:
Remove this assertion - if the code stops throwing exceptions, we know we broke BC :-)

Reply:
I suggest to stay with this one. Ok, it's not likely that the constant values of T_PUBLIC et al change. ;) But if the values were changed, the change could break the test (e.g. T_PUBLIC = 42).
BC would not be affected (in the particular context of this method).

Ok?

Reply:
You are testing something unrelated here, that's why I asked for removal.

Reply:
I see.  Now I make sure that the argument is invalid without asserting that.